bin/backfill_groupenvironment_first_seen

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python
+from __future__ import absolute_import
+
+from sentry.runner import configure
+configure()
+
+from operator import or_
+
+from django.db.models import Q
+
+from sentry.tagstore.v2.models import GroupTagValue
+from sentry.models import Environment, Group, GroupEnvironment
+
+CHUNK_SIZE = 10000
+
+
+def backfill_group_env():
+    ENV_CACHE = dict(Environment.objects.values_list('id', 'name'))
+
+    last_pk = None
+    qs = GroupEnvironment.objects.filter(first_seen__isnull=True).order_by('pk')
+    group_environments = list(qs[0:CHUNK_SIZE])
+
+    while group_environments:
+        last_pk = group_environments[-1].id
+        env_names = {
+            ENV_CACHE[ge.environment_id] for ge in group_environments
+            if ge.environment_id in ENV_CACHE
+        }
+
+        missing_envs = {ge.environment_id for ge in group_environments if ge.environment_id in ENV_CACHE}
+        # update cache in case some envs have been created since cache was populated
+        # not sure if this is actually necessary?
+        if missing_envs:
```

Comment:
Assuming we're writing `first_seen` on new rows this shouldn't be necessary, none of the data we're backfilling should change.

Reply:
yeah that's what i was thinking, but then second guessed myself